src/OrchardCore.Modules/OrchardCore.Users/Services/RecipeEnvironmentSuperUserProvider.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OrchardCore.Recipes.Services;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Users.Services
+{
+    public class RecipeEnvironmentSuperUserProvider : IRecipeEnvironmentProvider
+    {
+        private readonly ISiteService _siteService;
+        private readonly IUserService _userService;
+
+        public RecipeEnvironmentSuperUserProvider(ISiteService siteService, IUserService userService)
+        {
+            _siteService = siteService;
+            _userService = userService;
+        }
+
+        public async Task SetEnvironmentAsync(IDictionary<string, object> environment)
+        {
+            // When these have already been set by another provider, do not reset them.
+            if (!environment.ContainsKey("AdminUserId") && !environment.ContainsKey("AdminUsername"))
```

Comment:
Hmm, just not sure if a module depending on OC.Settings want to override the value but have its provider registered later, but yes weird to override the admin user name / id

Reply:
Yes, a little odd, but also because site settings returns a default value, when it has not yet been configured, i.e. My Orchard Application`

However the setup recipe environment provider actually runs later anyway, so this check is not necesary, to resolve the issue, (I added it before discovering the registration was later for the setup environment).
So I left the check in, in case registration orders change one day, without us noticing, for safety. 

But not needed so happy to remove it, or leave it there, just in case.

Or I could just put an order property on all of them, so it's more obvious what order they would be executed in?

Reply:
@deanmarcussen 

Are you talking about `RecipeEnvironmentFeatureProvider` that run later and that needs an higher precedence, if so you would need to remove this check, but if it may run first you would need to keep the checking.

That said the order may only depend on an alphabetic order if the related features don't depend on each other, so yes better to have an `Order` property to ensure some important ordering, but keeping the DI registration order for the same `Order` values. An higher precedence meaning running last with an higher `Order`.

Reply:
yeah that's what I'm meaning.

Cool. I agree, I'll remove this check. 

And place an `Order` property on the `IRecipeEnvironmentProvider` - makes it much more obvious what order things are happening in.

> An higher precedence meaning running last with an higher `Order`

Agree

Reply:
Done